bson-kotlinx/build.gradle.kts

```diff
@@ -38,19 +38,23 @@ description = "Bson Kotlinx Codecs"
 
 ext.set("pomName", "Bson Kotlinx")
 
+java { registerFeature("jsonSupport") { usingSourceSet(sourceSets["main"]) } }
```

Comment:
Has this functionality been tested with enabling an optional capability in a sample project?

Reply:
The pom produced has the correct scope:

```
    <dependency>
      <groupId>org.jetbrains.kotlinx</groupId>
      <artifactId>kotlinx-serialization-json</artifactId>
      <scope>compile</scope>
      <optional>true</optional>
    </dependency>
```